js/setup.js

```diff
@@ -1,80 +1,72 @@
-  const setupSimilarList = document.querySelector('.setup-similar-list');
-  const similarWizardTemplate = document.querySelector('#similar-wizard-template').content;
   const userNameInput = document.querySelector('.setup-user-name');
-  const wizardCoat = document.querySelector('.setup-wizard .wizard-coat');
-  const wizardCoatHidden = document.querySelector('input[name="coat-color"]');
-  const wizardEyes = document.querySelector('.setup-wizard .wizard-eyes');
-  const wizardEyesHidden = document.querySelector('input[name="eyes-color"]');
-  const wizardFireball = document.querySelector('.setup-fireball-wrap');
-  const wizardFireballHidden = document.querySelector('input[name="fireball-color"]');
-
-  const showCharacters = () => userDialog.querySelector('.setup-similar').classList.remove('hidden');
 
-  const addWizardToFragment = (fragment) => (wizard) => fragment.appendChild(renderWizard(wizard));
+  let coatColor = 'rgb(101, 137, 164)';
+  let eyesColor = 'black';
+  let wizards = [];
 
-  const renderWizard = ({name, colorCoat, colorEyes}) => {
-    const wizardElement = similarWizardTemplate.cloneNode(true);
-
-    wizardElement.querySelector('.setup-similar-label').textContent = name;
-    wizardElement.querySelector('.wizard-coat').style.fill = colorCoat;
-    wizardElement.querySelector('.wizard-eyes').style.fill = colorEyes;
-
-    return wizardElement;
-  };
+  const showCharacters = () => userDialog.querySelector('.setup-similar').classList.remove('hidden');
 
   showCharacters();
 
   const onUserNameInput = () => window.validation.validateUserName();
 
   userNameInput.addEventListener('input', onUserNameInput);
 
-  const onWizardCoatClick = () => {
-    wizardCoat.style.fill = window.colorize.getCoatColor();
-    wizardCoatHidden.value = wizardCoat.style.fill;
-  };
+  const getWizardRank = (wizard) => {
+    let rank = 0;
+
+    if (wizard.colorCoat === coatColor) {
+      rank = rank + 2;
+    }
 
-  wizardCoat.addEventListener('click', onWizardCoatClick);
+    if (wizard.colorEyes === eyesColor) {
+      rank = rank + 1;
+    }
 
-  const onWizardEyesClick = () => {
-    wizardEyes.style.fill = window.colorize.getEyesColor();
-    wizardEyesHidden.value = wizardEyes.style.fill;
+    return rank;
   };
 
-  wizardEyes.addEventListener('click', onWizardEyesClick);
+  const compareNames = (left, right) => {
+    if (left > right) {
+      return 1;
+    } else if (left < right) {
+      return -1;
+    } else {
+      return 0;
+    }
+  };
 
-  const onWizardFireballClick = () => {
-    const fireballColor = window.colorize.getFireballColor();
-    wizardFireball.style.backgroundColor = fireballColor;
-    wizardFireballHidden.value = fireballColor;
+  const updateWizards = () => {
+    window.render(wizards.sort((left, right) => {
+      let diff = getWizardRank(right) - getWizardRank(left);
+      if (diff === 0) {
+        diff = compareNames(left.name, right.name);
+      }
+      return diff;
+    }));
   };
```

Comment:
```suggestion
  const updateWizards = () => window.render(wizards..slice().sort(compareWizard);
```
делай slice, сорт мутирует начальный массив
```
const compareWizard
```